src/execution/window_executor.cpp

```diff
@@ -0,0 +1,1280 @@
+    : WindowExecutor(wexpr, context, payload_count, partition_mask, order_mask) {
+}
+
+unique_ptr<WindowExecutorState> WindowRankExecutor::GetExecutorState() const {
+	return make_uniq<WindowPeerState>(wexpr, context, payload_count, partition_mask, order_mask);
+}
+
+void WindowRankExecutor::EvaluateInternal(WindowExecutorState &lstate, Vector &result, idx_t count,
+                                          idx_t row_idx) const {
+	auto &lpeer = lstate.Cast<WindowPeerState>();
+	auto partition_begin = FlatVector::GetData<const idx_t>(lpeer.bounds.data[PARTITION_BEGIN]);
+	auto peer_begin = FlatVector::GetData<const idx_t>(lpeer.bounds.data[PEER_BEGIN]);
+	auto rdata = FlatVector::GetData<int64_t>(result);
+
+	//	Reset to "previous" row
+	lpeer.rank = (peer_begin[0] - partition_begin[0]) + 1;
+	lpeer.rank_equal = (row_idx - peer_begin[0]);
+
+	for (idx_t i = 0; i < count; ++i, ++row_idx) {
+		lpeer.NextRank(partition_begin[i], peer_begin[i], row_idx);
+		rdata[i] = lpeer.rank;
+	}
+}
+
+WindowDenseRankExecutor::WindowDenseRankExecutor(BoundWindowExpression &wexpr, ClientContext &context,
+                                                 const idx_t payload_count, const ValidityMask &partition_mask,
+                                                 const ValidityMask &order_mask)
+    : WindowExecutor(wexpr, context, payload_count, partition_mask, order_mask) {
+}
+
+unique_ptr<WindowExecutorState> WindowDenseRankExecutor::GetExecutorState() const {
+	return make_uniq<WindowPeerState>(wexpr, context, payload_count, partition_mask, order_mask);
+}
+
+void WindowDenseRankExecutor::EvaluateInternal(WindowExecutorState &lstate, Vector &result, idx_t count,
+                                               idx_t row_idx) const {
+	auto &lpeer = lstate.Cast<WindowPeerState>();
+	auto partition_begin = FlatVector::GetData<const idx_t>(lpeer.bounds.data[PARTITION_BEGIN]);
+	auto peer_begin = FlatVector::GetData<const idx_t>(lpeer.bounds.data[PEER_BEGIN]);
+	auto rdata = FlatVector::GetData<int64_t>(result);
+
+	//	Reset to "previous" row
+	lpeer.rank = (peer_begin[0] - partition_begin[0]) + 1;
+	lpeer.rank_equal = (row_idx - peer_begin[0]);
+
+	//	The previous dense rank is the number of order mask bits in [partition_begin, row_idx)
+	lpeer.dense_rank = 0;
+
+	auto order_begin = partition_begin[0];
+	idx_t begin_idx;
+	idx_t begin_offset;
+	order_mask.GetEntryIndex(order_begin, begin_idx, begin_offset);
+
+	auto order_end = row_idx;
+	idx_t end_idx;
+	idx_t end_offset;
+	order_mask.GetEntryIndex(order_end, end_idx, end_offset);
+
+	//	If they are in the same entry, just loop
+	if (begin_idx == end_idx) {
+		const auto entry = order_mask.GetValidityEntry(begin_idx);
+		for (; begin_offset < end_offset; ++begin_offset) {
+			lpeer.dense_rank += order_mask.RowIsValid(entry, begin_offset);
+		}
+	} else {
+		// Count the ragged bits at the start of the partition
+		if (begin_offset) {
```

Comment:
Same here - this appears to be uncovered

Reply:
I have not been able to generate a case for this. Basically, it would happen when you have two partitions in the same hash group where the boundary was not in the first chunk. That would require a lot of large partitions, and `test/sql/window/test_dense_rank.test` was my unsuccessful attempt to trigger this.